File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        reverse(matrix.begin(),matrix.end());
        int n=matrix.size();
        for(int i=0;i<n;i++)
        {

           
                for(int j=0;j<=i;j++)
                {
                    swap(matrix[i][j],matrix[j][i]);
                }
            
        }
    //     int col=matrix[0].size();
    //     for(int i=0;i<n;i++)
    //     {
    //         for(int j=0;j<col/2;j++)
    //         {
    //             swap(matrix[i][j],matrix[i][col-1-j]);
    //         }

    //     }
        
    }
};

File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        n=len(matrix)

        ans= [[0 for _ in range(n)] for _ in range(n)]
        for i in range(n):
            for j in range(n):
                ans[j][n-i-1]=matrix[i][j]
        for i in range(n):
            for j in range(n):
                matrix[i][j] = ans[i][j]
        